src/components/LazyMotion/index.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from "react"
+import { useEffect, useRef, useState } from "react"
+import { LazyContext } from "../../context/LazyContext"
+import { loadFeatures } from "../../motion/features/definitions"
+import { FeatureBundle, LazyFeatureBundle } from "../../motion/features/types"
+import { CreateVisualElement } from "../../render/types"
+import { LazyProps } from "./types"
+
+/**
+ * Used in conjunction with the `m` component to reduce bundle size.
+ *
+ * `m` is a version of the `motion` component that only loads functionality
+ * critical for the initial render.
+ *
+ * `LazyMotion` can then be used to either synchronously or asynchronously
+ * load animation and gesture support.
+ *
+ * ```jsx
+ * // Synchronous loading
+ * import { LazyMotion, m, domAnimations } from "framer-motion"
+ *
+ * function App() {
+ *   return (
+ *     <LazyMotion features={domAnimations}>
+ *       <m.div animate={{ scale: 2 }} />
+ *     </LazyMotion>
+ *   )
+ * }
+ *
+ * // Asynchronous loading
+ * import { LazyMotion, m } from "framer-motion"
+ *
+ * function App() {
+ *   return (
+ *     <LazyMotion features={() => import('./path/to/domAnimations')}>
+ *       <m.div animate={{ scale: 2 }} />
+ *     </LazyMotion>
+ *   )
+ * }
+ * ```
+ *
+ * @public
+ */
+export function LazyMotion({ children, features, strict = false }: LazyProps) {
+    const [, setIsLoaded] = useState(!isLazyBundle(features))
+    const loadedRenderer = useRef<undefined | CreateVisualElement<any>>(
+        undefined
+    )
+
+    /**
+     * If this is a synchronous load, load features immediately
+     */
+    if (!isLazyBundle(features)) {
+        const { renderer, ...loadedFeatures } = features
+        loadedRenderer.current = renderer
+        loadFeatures(loadedFeatures)
+    }
+
+    useEffect(() => {
+        if (isLazyBundle(features)) {
+            features().then(({ renderer, ...loadedFeatures }) => {
+                loadFeatures(loadedFeatures)
+                loadedRenderer.current = renderer
+                setIsLoaded(true)
+            })
```

Comment:
Any chance you need to catch here or is it up to the user to make sure that the features promise is handled accordingly?

Reply:
(I think it's fine if it's the latter)

Reply:
Ah yeah this is actually why I didn't handle the loading in the initial iteration. I'll have a quick ponder.

Reply:
Oh yeah I've had a play around and its probably better to figure this out within the dynamic function. A user could conceivably create a custom retry function here.

Reply:
Makes sense to me 👌